pkgs/desktops/gnome-3/extensions/update-extensions.py

```diff
@@ -0,0 +1,245 @@
+        else:
+            package_name_registry[shell_version][pname] = [uuid]
+
+    return {
+        "uuid": uuid,
+        "name": sanitize_string(extension["name"]),
+        "pname": pname,
+        "description": sanitize_string(extension["description"]),
+        "link": "https://extensions.gnome.org" + extension["link"],
+        "shell_version_map": shell_version_map,
+    }
+
+
+def fetch_extensions() -> List[Dict[str, Any]]:
+    """
+    Fetch the extensions. The extensions database is scraped by using the search query. We simply go over it
+    page by page until we hit a non-full page or a 404 error.
+
+    The returned list is sorted by the age of the extension, in order to be deterministic.
+    """
+    page = 0
+    extensions = []
+    while True:
+        page += 1
+        logging.info("Scraping page " + str(page))
+        try:
+            with urllib.request.urlopen(
+                f"https://extensions.gnome.org/extension-query/?n_per_page=25&page={page}"
+            ) as url:
+                data = json.loads(url.read().decode())["extensions"]
+                responseLength = len(data)
+
+                for extension in data:
+                    extensions += [extension]
+
+                # If our page isn't "full", it must have been the lasts one
+                if responseLength < 25:
+                    logging.debug(
+                        f"\tThis page only has {responseLength} entries, so it must be the last one."
+                    )
+                    break
+        except urllib.error.HTTPError:
+            # Assume this error is a 404. We hit the last page; we're done.
+            break
+
+    extensions.sort(key=itemgetter("pk"))
+    return extensions
+
+
+if __name__ == "__main__":
+    raw_extensions = fetch_extensions()
+
+    logging.info(f"Downloaded {len(raw_extensions)} extensions. Processing …")
+    processed_extensions: List[Dict[str, Any]] = []
+    for raw_extension in raw_extensions:
+        processed_extension = process_extension(raw_extension)
+        if processed_extension:
+            processed_extensions += [processed_extension]
+
+    logging.info(
+        f"Done. Writing results to extensions.json ({len(processed_extensions)} extensions in total)"
+    )
+
+    with open("extensions.json", "w") as out:
+        # Manually pretty-print the outer level, but then do one compact line per extension
+        # This allows for the diffs to be manageable (one line of change per extension) despite their quantity
+        for index, extension in enumerate(processed_extensions):
+            if index == 0:
+                out.write("[   ")
+            else:
+                out.write(",   ")
+            json.dump(extension, out)
+            out.write("\n")
+        out.write("]\n")
+
+    with open("extensions.json", "r") as out:
+        # Check that the generated file actually is valid JSON, just to be sure
+        json.load(out)
+
+    logging.info(
```

Comment:
Do you maybe want to print the number of collisions?

Maybe even print a warning if there was a collision.json before and it differs?

Reply:
I had this code before, but storing that information in the VCS-tracked `collisions.json` made it redundant so I removed it.

Reply:
I am not talking about storing any more information in the file.
I just suggest printing a warning, when the file changes. That’s just about the ergonomics of using this script though and you are welcome to look for changes manually using git.